crates/bevy_reflect/src/from_reflect.rs

```diff
@@ -25,7 +27,7 @@ use crate::{FromType, PartialReflect, Reflect};
     message = "`{Self}` does not implement `FromReflect` so cannot be created through reflection",
     note = "consider annotating `{Self}` with `#[derive(Reflect)]`"
 )]
-pub trait FromReflect: Reflect + Sized {
+pub trait FromReflect: Any + CastPartialReflect + Sized {
```

Comment:
Why this new `Any` bound?

Reply:
`Reflect` has `Any` as a super trait, so I just kept it here to help reduce the scope of the breakages